greenplumpython/table.py

```diff
@@ -52,6 +55,77 @@ def select(self, target_list: Iterable) -> "Table":
             parents=[self],
         )
 
+    def _join(
+        self,
+        other: "Table",
+        targets: List = [],
+        how: str = "",
+        on_str: str = "",
+    ) -> "Table":
+        # FIXME : Raise Error if target columns don't exist
+        # FIXME : Same column name in both table
+        select_str = ",".join([str(target) for target in targets]) if targets != [] else "*"
+        return Table(
+            f"""
+                SELECT {select_str} 
+                FROM {self.name} {how} {other.name}
+                {str(on_str)}  
+            """,
+            parents=[self, other],
+        )
+
+    def inner_join(
+        self,
+        other: "Table",
+        cond: expr.Expr,
+        targets: List = [],
+    ):
+        on_str = " ".join(["ON", str(cond)])
+        return self._join(other, targets, "INNER JOIN", on_str)
+
+    def left_join(
+        self,
+        other: "Table",
+        cond: expr.Expr,
+        targets: List = [],
+    ):
+        on_str = " ".join(["ON", str(cond)])
+        return self._join(other, targets, "LEFT JOIN", on_str)
+
+    def right_join(
+        self,
+        other: "Table",
+        cond: expr.Expr,
+        targets: List = [],
+    ):
+        on_str = " ".join(["ON", str(cond)])
+        return self._join(other, targets, "RIGHT JOIN", on_str)
+
+    def full_outer_join(
+        self,
+        other: "Table",
+        cond: expr.Expr,
+        targets: List = [],
+    ):
+        on_str = " ".join(["ON", str(cond)])
+        return self._join(other, targets, "FULL JOIN", on_str)
+
+    def natural_join(
+        self,
+        other: "Table",
+        targets: List = [],
+    ):
+        on_str = ""
+        return self._join(other, targets, "FULL JOIN", on_str)
```

Comment:
This is still wrong. `FULL JOIN` -> `NATURAL JOIN`. And you don't have right tests cover this.

It is strange that my old comments disappeared.
https://github.com/greenplum-db/GreenplumPython/pull/49/files/61eb1e0920fdf833eced6d2429965dd9266d9856#r894127757